pandas/tests/test_categorical.py

```diff
@@ -4549,6 +4549,13 @@ def test_concat_categorical(self):
                             'h': [None] * 6 + cat_values})
         tm.assert_frame_equal(res, exp)
 
+    def test_rank_categorical(self):
```

Comment:
good point @TomAugspurger 

please also exericse the ``na_option`` and ``ascending`` options as well.

Use other tests as models in tests_anlaytics for this type of testing.